src/deepsparse/pipeline.py

```diff
@@ -300,24 +291,22 @@ def __call__(self, *args, **kwargs) -> BaseModel:
                 f"Outputs of {self.__class__} must be instances of "
                 f"{self.output_schema} found output of type {type(pipeline_outputs)}"
             )
-        timer.stop(InferencePhases.POST_PROCESS)
-        timer.stop(InferencePhases.TOTAL_INFERENCE)
-
+        self.inference_timer.stop_inference_stage(InferenceStages.POST_PROCESS)
         self.log(
             identifier="pipeline_outputs",
             value=pipeline_outputs,
             category=MetricCategories.DATA,
         )
-        self.log(
-            identifier=f"{SystemGroups.PREDICTION_LATENCY}/{InferencePhases.POST_PROCESS}_seconds",  # noqa E501
-            value=timer.time_delta(InferencePhases.POST_PROCESS),
-            category=MetricCategories.SYSTEM,
-        )
-        self.log(
-            identifier=f"{SystemGroups.PREDICTION_LATENCY}/{InferencePhases.TOTAL_INFERENCE}_seconds",  # noqa E501
-            value=timer.time_delta(InferencePhases.TOTAL_INFERENCE),
-            category=MetricCategories.SYSTEM,
-        )
+
+        # ------ INFERENCE FINALIZATION ------
+        self.inference_timer.stop_inference_stage(InferenceStages.TOTAL_INFERENCE)
+        logging_times = self.extract_loggable_times()
+        for time_key, time_value in logging_times.items():
+            self.log(
```

Comment:
Note: we will be delaying three out of four logging events, which may be confusing to the user (instead of logging immediately after each inference stage, we end until the end of the forward pass and only then log all the latency logs in one go, this is not optimal)

Reply:
for timings logging after inference completes will be permissible for now since data logging is still immediately logged for debugging